RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -430,33 +415,16 @@ private string BuildPlugin(RoslynPluginDefinition definition, string baseTempDir
             return fullJarPath;
         }
 
-        private void AddRoslynMetadata(string baseTempDirectory, PluginBuilder builder, RoslynPluginDefinition definition)
+        private void AddRoslynMetadata(RoslynPluginJarBuilder builder, RoslynPluginDefinition definition, string repositoryId)
         {
-            string sourcesDir = Utilities.CreateSubDirectory(baseTempDirectory, "src");
-            this.logger.LogDebug(UIResources.APG_CreatingRoslynSources, sourcesDir);
-
-            SourceGenerator.CreateSourceFiles(typeof(AnalyzerPluginGenerator).Assembly, RoslynResourcesRoot, sourcesDir, new Dictionary<string, string>());
-
-            string[] sourceFiles = Directory.GetFiles(sourcesDir, "*.java", SearchOption.AllDirectories);
-            Debug.Assert(sourceFiles.Any(), "Failed to correctly unpack the Roslyn analyzer specific source files");
-
-            foreach (string sourceFile in sourceFiles)
-            {
-                builder.AddSourceFile(sourceFile);
-            }
-
-            builder.SetSourceCodeTokenReplacement(PackageId_Token, definition.PackageId);
-            builder.SetSourceCodeTokenReplacement(PackageVersion_Token, definition.PackageVersion);
-            builder.SetSourceCodeTokenReplacement(AnalyzerId_Token, definition.PackageId);
-            builder.SetSourceCodeTokenReplacement(RuleNamespace_Token, definition.PackageId);
-            builder.SetSourceCodeTokenReplacement(StaticResourceName_Token, definition.StaticResourceName);
-            builder.SetSourceCodeTokenReplacement(PluginKey_Token, definition.Manifest.Key);
-            builder.SetSourceCodeTokenReplacement(PluginVersion_Token, definition.Manifest.Version);
-
-            foreach (string extension in Extensions)
-            {
-                builder.AddExtension(extension);
-            }
+            builder.SetPluginProperty(repositoryId + ".nuget.packageId", definition.PackageId);
+            builder.SetPluginProperty(repositoryId + ".nuget.packageVersion", definition.PackageVersion);
+
+            builder.SetPluginProperty(repositoryId + ".analyzerId", definition.PackageId);
+            builder.SetPluginProperty(repositoryId + ".ruleNamespace", definition.PackageId);
+            builder.SetPluginProperty(repositoryId + ".staticResourceName", definition.StaticResourceName);
+            builder.SetPluginProperty(repositoryId + ".pluginKey", definition.Manifest.Key);
+            builder.SetPluginProperty(repositoryId + ".pluginVersion", definition.Manifest.Version);
```

Comment:
The "SetPluginProperty" replaces the old code which injected the property values into source code that was then compiled. The properties are now written to the new configuration.xml file which is then embedded in the jar.